src/toolkit/matl_buy_policy.cc

```diff
@@ -199,10 +241,33 @@ void MatlBuyPolicy::AcceptMatlTrades(
   for (it = resps.begin(); it != resps.end(); ++it) {
     rsrc_commods_[it->second] = it->first.request->commodity();
     LGH(INFO3) << "got " << it->second->quantity() << " kg of "
-               << it->first.request->commodity();
+               << it->first.request->commodity()  << std::endl;
     buf_->Push(it->second);
   }
 }
 
+void MatlBuyPolicy::SetNextActiveTime() {
+  next_active_end_ = active_dist_->sample() + manager()->context()->time();
+  return;
+};
+
+void MatlBuyPolicy::SetNextDormantTime() {
+  if (next_dormant_end_ < 0) {std::cerr << "dormant length -1, always active" << std::endl;}
+  else {
+    std::cerr << "setting next dormant time" << std::endl;
+    next_dormant_end_ = dormant_dist_->sample() + next_active_end_;
+  }
+  return;
+}
+
+double MatlBuyPolicy::SampleRequestSize() {
+  double random_request_size_ = size_dist_->sample();
+  if (random_request_size_ > 1) {
+    random_request_size_ = random_request_size_ / size_dist_->max();
+  }
```

Comment:
This seems like it will give inconsistent behavior - if `size_dist_->max() > 1` it's possible to get a random sample that is less than 1 that will NOT be normalized by `size_dist_->max()` and then another sample that is greater than 1 that WILL be normalized.  

Maybe the condition should be to divide by `size_dist_->max()` if `size_dist_->max() > 1`??

Reply:
You know, as I'm thinking about this, it may make sense for all reasonable user use cases (like myself, encountering this while writing the tests). You're right that my original version is inconsistent in any case. Your version makes sense In the context of uniform distribution, which is required to have a max, but less so for normal, which is not. For a normal dist, the max defaults to the numeric limit. Therefore, a default normal dist, which only sets mean and stddev, will constantly be scaled against the largest value possible, resulting in a zero request size. 
Instead of adding fancy handling here, it may be simplest to just say that if 
```suggestion
  if (random_request_size_ > 1) {
    random_request_size_ = 1.0;
  }
```
aka the simplest artificial max

Reply:
Two things:

1) This will skew the sampling of the distribution.  What if someone has a Normal Distribution with a max of 1.1?  The sampling routine will reject/resample until it gets a value less than 1.1, but this will simple change the value rather than resample.  That's an inconsistent handling of your stochastic variable.

2) Therefore, it might make more sense to require all distributions used for sample size to have a max that is less than or equal to 1?  This could be tested when the policy is initialized and cause an error (or automatic correction) when violated.